### DISCUSSION TITLE
Feature extraction for time series forecasting

### DISCUSSION THREAD

**Discussion by KingKai69** [blue-yonder/tsfresh#1059] [Q&A]
Hey all,

currently i am doing my master thesis and i am dealing with time series forecasting with regression models like XGBoost. During my research i found tsfresh.

I have a single time series of a KPI and a external regresor (market data) to predict future values of the KPI.

My question is if the extracted_features() respectively extract_relevant_features() function also work for time series forecasting? If i use the named functions the extracted features df contrains just one row with X columns, but in the end i need a feature value for each timestamp. Do those functions work for forecasting problems or should i rather use the rolling mechanism described here https://tsfresh.readthedocs.io/en/latest/text/forecasting.html

As i understood the rolling mechanism just created kind of sliding windows based on a timeseries and the extract relevant features function extracts "new" features. Can anyone also clarify that?

Best,
Kai

**Comment by nils-braun** [comment]
Hi @KingKai69! 
Exactly, your understanding is correct. You first roll into slides and then extract the features from each slide - as explained in the documentation you linked 👍
